src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts

```diff
@@ -45,8 +49,29 @@ export class DokobitAuthenticationProvider implements AuthenticationProvider {
 
     return {
       code: sessionStatus.code,
-      firstName: sessionStatus.name,
-      lastName: sessionStatus.surname,
+      profile: mapToProfile(sessionStatus),
     };
   }
 }
+
+function mapToProfile(sessionStatus: DokobitSessionStatus): Profile {
+  var id = getAndValidateEstonianIdCode(sessionStatus);
+  id.validate();
```

Comment:
We're already validating in `getAndValidateEstonianIdCode`, is this necessary?

Reply:
oh.. missed this... good spot